include/dlaf/communication/error.h

```diff
@@ -0,0 +1,46 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2019, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#include <cstdio>
+#include <iostream>
+#include <memory>
+
+#include <mpi.h>
+
+namespace dlaf {
+namespace internal {
+
+/// MPI provides two predefined error handlers
+///
+/// - `MPI_ERRORS_ARE_FATAL` (the default ?): causes MPI to abort
```

Comment:
Yes this is the default, and the reason why MPI calls are usually not checked for error code.
Since MPI errors are not recoverable the use of MPI_ERRORS_RETURN is useful only if the application can dump a snapshot and use it when it is restarted.
However most of the applications that use checkpoints do not use this approach to avoid to have to deal with different entry point and inconsistent data among ranks. They usually just always write a checkpoint file in specific points of the code.
Do we really want to put extra checks on MPI calls?